aspnet/mvc/views/overview.rst

```diff
@@ -0,0 +1,97 @@
+
+A sample view file (*About.cshtml*):
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/Views/Home/About.cshtml
+   :language: html
+
+*Razor* code is denoted by the ``@`` symbol. C# statements are run within Razor code blocks set off by curly braces (``{`` ``}``), such as the assignment of "About" to the ``ViewData["Title"]`` element shown above. Razor can be used to display values within HTML by simply referencing the value with the ``@`` symbol, as shown within the ``<h2>`` and ``<h3>`` elements above.
+
+This view focuses on just the portion of the output for which it is responsible. The rest of the page's layout, and other common aspects of the view, are specified elsewhere. Learn more about :doc:`layout and shared view logic <layout>`.
+
+How do Controllers Specify Views?
+---------------------------------
+
+Views are typically returned from actions as a :dn:cls:`~Microsoft.AspNetCore.Mvc.ViewResult`. Your action method can create and return a ``ViewResult`` directly, but more commonly if your controller inherits from :dn:cls:`~Microsoft.AspNetCore.Mvc.Controller`, you'll simply use the ``View`` helper method, as this example demonstrates:
+
+*HomeController.cs*
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/Controllers/HomeController.cs
+  :language: c#
+  :lines: 16-21
+  :emphasize-lines: 5
+  :dedent: 4
+  
+The ``View`` helper method has several overloads to make returning views easier for app developers. You can optionally specify a specific view to return, as well as a model to pass to the view. With no parameters, the default convention of using a view name that corresponds to the action name is used.
+
+When this action returns, the *About.cshtml* view shown above is rendered:
+
+.. image:: overview/_static/about-page.png
+
+View Discovery
+^^^^^^^^^^^^^^
+
+When an action returns a view, a process called *view discovery* takes place. This process determines which view file will be used. There are several scenarios, based on how the view was specified from the action method.
+
+``return View();``
+    The most common approach. In this case the framework will use a view with the same name as the action (with a *.cshtml* extension) located in a folder named the same as the controller. Otherwise, it will look in the *Shared* folder for a view with a name corresponding to the action name.
+
+``return View("List");``
+    Using this syntax, a view with the provided name (and a *.cshtml* extension) is used instead of the current action name. Otherwise, the behavior is as above.
+
+``return View("Views/Home/About.cshtml");``
+    Specify the path from the application root to the view. Using this syntax, the *.cshtml* extension must be specified. The path can optionally start with "/" or "~/".
```

Comment:
BTW I wouldn't say "magic string". A magic string is a string that's opaque, whereas this is nothing like that. It's just a path string.